pkg/cosign/verify.go

```diff
@@ -807,3 +813,12 @@ func correctAnnotations(wanted, have map[string]interface{}) bool {
 	}
 	return true
 }
+
+func correctCertExtensions(wanted, have map[string]string) bool {
+	for k, v := range wanted {
+		if have[k] != v {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
I'm not sure about this function. Kyverno ignores empty-string values (same as the extension would not be required) and respects shortened values or wildcards. However, it is of course not mandatory that cosign will do the same here. WDYT?

Reply:
It is the function that makes the matching between the actual one and the expected one to decide whether claims are equal, I couldn't understand your comment :/

Reply:
Yes, correct. What I mean is, that this function checks for strict equality, wildcards and shortened values are not supported.
E.g. `cosign verify aquasec/trivy:0.24.3 --cert-extensions githubWorkflowRepository=aquasecurity/*` won't work (with or without asterisk). This is how Kyverno would validate: https://github.com/kyverno/kyverno/blob/main/pkg/cosign/cosign.go#L419-L421

Reply:
It is not required to make the validation equal to the Kyverno logic, just want to mention the difference 😉